widget/embedded/src/components/SwapsGroup/SwapsGroup.tsx

```diff
@@ -11,11 +16,52 @@ import {
 import { getContainer } from '../../utils/common';
 import { formatTooltipNumbers, numberToString } from '../../utils/numbers';
 
-import { Group, groupStyles, SwapList, Time } from './SwapsGroup.styles';
+import {
+  Group,
+  groupStyles,
+  SwapItemContainer,
+  SwapList,
+  Time,
+} from './SwapsGroup.styles';
+
+const ITEMS_PER_PAGE = 10;
 
 export function SwapsGroup(props: PropTypes) {
   const { list, onSwapClick, groupBy, isLoading } = props;
-  const groups = groupBy ? groupBy(list) : [{ title: 'History', swaps: list }];
+  const [currentGroupCounts, setCurrentGroupCounts] = useState<number[]>([]);
+  const loadedItems = useRef(0);
+  const { swaps, groupCounts, groups } = groupBy(list);
+
+  const calculateGroupsSoFar = useCallback(
```

Comment:
I don't understand what is this. Could you please describe?

Reply:
The calculateGroupsSoFar Slices the total groups into the groups which contain the items so far. For example, if you have [10, 10, 10, 10] groups in total, slicing them to 23 will result in [10, 10, 3] :
https://virtuoso.dev/grouped-with-load-on-demand/